### PR TITLE
exult: 1.4.9rc1 -> 1.6

### PR DESCRIPTION
**Pull request by patrl**
###### Motivation for this change

Update exult to verion 1.6.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Some notes:
- As far as I can see, the patch is now unnecessary, so I've removed it. 
- exult is now built against SDL2.
- I've removed `makeFlags`, since it was resulting in a nested dir structure in the nix store.

pinging: @edolstra

### CODE REVIEW

**Comment by patrl** [comment]
This should be a pretty straightforward PR. Is there anything I can do to get this moving?

**Comment by patrl** [comment]
No worries! Commits squashed!!

File: pkgs/games/exult/default.nix [hunk 1]
```diff
@@ -7,32 +7,27 @@ let
...
 
 stdenv.mkDerivation rec {
-  name = "exult-1.4.9rc1";
+  name = "exult-1.6";
 
   src = fetchurl {
     url = "mirror://sourceforge/exult/${name}.tar.gz";
-    sha256 = "0a03a2l3ji6h48n106d4w55l8v6lni1axniafnvvv5c5n3nz5bgd";
+    sha256 = "1dm27qkxj30567zb70q4acddsizn0xyi3z87hg7lysxdkyv49s3s";
+    # date = 2020-06-02T10:50:06-0400;
```
Comment (Lassulus) [inline pkgs/games/exult/default.nix:21]: why post the date here?

Comment (patrl) [reply pkgs/games/exult/default.nix:21]: This is an artifact John Wiegley's nix-update.el script. I can remove it if necessary.

Comment (Lassulus) [reply pkgs/games/exult/default.nix:21]: ah, IC. Not sure the next person will bump the timestamp if it's not you, so removing would be better IMHO

Comment (patrl) [reply pkgs/games/exult/default.nix]: done

**Pull request merged**